lib/catalog/order_item_sanitized_parameters.rb

```diff
@@ -69,22 +64,52 @@ def mask_value?(field)
     end
 
     def fields
-      service_plan_schema.dig(:schema, :fields) || []
+      @fields ||= service_plan_schema.with_indifferent_access.dig(:schema, :fields)
     end
 
     def service_plan_does_not_exist?
       service_plan_ref.nil?
     end
 
+    def sanitize_value(key, value)
+      field = fields.find { |f| f[:name] == key }
+      return value unless field[:isSubstitution]
+
+      str_val = substitute(value)
+      Rails.logger.warn("Substitution result for expression #{value} is blank") if str_val.blank?
```

Comment:
@bzwei Should we add a Progress Message when we get a warning
like here
https://github.com/RedHatInsights/catalog-api/blob/96e5135f582fce8d1eb8e891a3175b88c09dd2fe/app/models/approval_request.rb#L8